Marlin/Marlin_main.cpp

```diff
@@ -5412,11 +5411,14 @@ void home_all_axes() { gcode_G28(); }
         }
 
         stepper.synchronize();
-
-        home_all_axes();
+        endstops.enable(true);
```

Comment:
In general, after enabling endstops to do something, when done the function to call is `endstops.not_homing()` which will set them back to their global non-homing state.